src/app/ui/editor/standby_state.cpp

```diff
@@ -747,6 +756,13 @@ void StandbyState::transformSelection(Editor* editor, MouseMessage* msg, HandleT
     return;
   }
 
+  if (layer && !layer->isEditableHierarchy()) {
```

Comment:
There are too much duplicated code calling `transformSelection()` that already verifies the lock/visible state of the layer, maybe we can move all those verifications inside this function:
```
      if (!layer->isEditableHierarchy()) {
        StatusBar::instance()->showTip(
          1000, fmt::format("Layer '{}' is locked", layer->name()));
        return true;
      }
      else if (!layer->isVisibleHierarchy()) {
        StatusBar::instance()->showTip(
          1000, fmt::format("Layer '{}' is hidden", layer->name()));
        return true;
      }
```